### PR TITLE
Bitfield-ify UART intClear register.

### PR DESCRIPTION
**Pull request by jlebar**
<git-pr-chain>

#### Commits in this PR
1. Bitfield-ify UART intClear register.

#### [PR chain](https://github.com/jlebar/git-pr-chain)
1. 👉 #342 Bitfield-ify UART intClear register. 👈 **YOU ARE HERE**
1. #343 C++ nits in uart_dma.
1. #344 Templatize CircularBuffer's datatype.

</git-pr-chain>

### CODE REVIEW

File: controller/lib/hal/hal_stm32_regs.h [hunk 1]
```diff
@@ -222,7 +222,29 @@ struct UART_Regs {
...
+      REG idlecf : 1; // idle line detected clear flag
+      REG rsvd : 1;
+      REG tccf : 1;    // transmission complete clear flag
+      REG tcbgtcf : 1; // transmission completed before guard time clear fl.
+      REG lbdcf : 1;   // LIN break detection clear flag
+      REG ctscf : 1;   // CTS clear flag
+      REG rsvd2 : 1;
+      REG rtocf : 1; // receiver timeout clear flag
+      REG eobcf : 1; // end of block clear flag
+      REG rsvd3 : 3;
```
Comment (Miceuz) [inline controller/lib/hal/hal_stm32_regs.h:240]: 4 bits here!

File: controller/lib/hal/hal_stm32_regs.h [hunk 2]
```diff
@@ -222,7 +222,29 @@ struct UART_Regs {
...
+      REG cmcf : 1; // character match clear flag
+      REG rsvd4 : 2;
+      REG wucf : 1; // wakeup from stop mode clear flag
+      REG rsvd5 : 10;
```
Comment (Miceuz) [inline controller/lib/hal/hal_stm32_regs.h:244]: 11 bits here!

File: controller/lib/hal/hal_stm32_regs.h (hunk 1 above)
Comment (jlebar) [reply controller/lib/hal/hal_stm32_regs.h:240]: Yikes, thanks for catching this.

**Comment by jlebar** [comment]
Thanks for catching the issues here and reviewing the other PRs, @Miceuz.

**Pull request merged**